R2RDump fixes for dumping method instance entrypoint table

trylek opened the pull request:
While investigating Michal's unit test demonstrating a bug in the
CPAOT compiler I found out that R2RDump is broken in multiple
aspects w.r.t. instantiated methods. Parsing of the method
signatures in the instance entrypoint table was just broken
and the R2RMethod was duplicating parts of the signature decoding
and name formatting process.

In addition to fixing these issues, I changed default method dump
to be ordered by signatures, not RID's, as that makes comparing
CPAOT and Crossgen R2R dumps much easier. This behavior can be
overridden by specifying the new --unordered command-line switch
which turns on the old behavior i.e. RID-based method ordering.

Thanks

Tomas

jkotas commented:
> `-unordered` command-line switch

I think the output normalizations like ordering, or omitting absolute addresses should be opt-in. The other (unmanaged) binary dumper tools do not normalize by default. I think it would be non-intuitive for R2R dump to normalize by default.

trylek commented:
I have rebased the change against the current master, I have addressed JanK's feedback by reversing the polarity of the switch and I added a bit of extra cleanup w.r.t. method signature formatting and passing around the various dump switches.
